static/scripts/audit-report/audit.ts

```diff
@@ -315,244 +364,93 @@ class QueueSet {
-    }
-  } else {
-    console.log("URL not found, skipping");
+  } catch (error) {
+    console.error(error);
+    throw error;
   }
 
-  return isFound;
+  return permitList;
 }
 
 async function gitFetcher(repoUrls: GitHubUrlParts[]) {
   if (isGit) {
     try {
-      const issuesPromises = repoUrls.map((repoUrl) => getIssuesForRepo(repoUrl.owner, repoUrl.repo));
-      const allIssues = await Promise.all(issuesPromises);
+      const permitsPromises = repoUrls.map((repoUrl) => getPermitsForRepo(repoUrl.owner, repoUrl.repo));
+      const allPermits = await Promise.all(permitsPromises);
 
-      for (let i = 0; i < allIssues.length; i++) {
-        const issues = allIssues[i];
-        issueList.push(...issues);
-        console.log(`Fetched ${issues.length} issues for repository ${repoUrls[i].owner}/${repoUrls[i].repo}`);
+      for (let i = 0; i < allPermits.length; i++) {
+        const issues = allPermits[i];
+        permitList.push(...issues);
+        console.log(`Fetched ${issues.length} permits for repository ${repoUrls[i].owner}/${repoUrls[i].repo}`);
       }
-
       isGit = false;
       finishedQueue.mutate("isGit", true);
-      await commentFetcher();
-    } catch (error: unknown) {
-      console.error("Error fetching issues:", error);
-    }
-  }
-}
-
-async function getIssuesForRepo(owner: string, repo: string) {
-  const issueList: ListForRepoDataItem[] = [];
-
-  const octokit = new rateOctokit({
-    auth: GITHUB_PAT,
-    throttle: {
-      onRateLimit: (retryAfter, options) => {
-        return primaryRateLimitHandler(retryAfter, options as RateLimitOptions);
-      },
-      onSecondaryRateLimit: (retryAfter, options) => {
-        return secondaryRateLimitHandler(retryAfter, options as RateLimitOptions);
-      },
-    },
-  });
-
-  const isIEF = true;
-
-  while (isIEF) {
-    try {
-      const { data } = await octokit.rest.issues.listForRepo({
-        owner,
-        repo,
-        state: "closed",
-        per_page: offset,
-        page: gitPageNumber,
-      });
-
-      if (data.length === 0) break;
-
-      const issues = data.filter((issue) => !issue.pull_request && issue.comments > 0);
-      const { isIEF: isIEF2, issueList: processed } = await processIssues(isIEF, issueList, issues);
-
-      issueList.push(...processed);
-
-      if (!isIEF2) {
-        break;
-      }
-    } catch (error: unknown) {
-      console.error(error);
-      throw error;
-    }
-  }
-
-  return issueList;
-}
-
-async function processIssues(isIEF: boolean, issueList: ListForRepoDataItem[], issues: ListForRepoDataItem[]) {
-  if (issues.length > 0) {
-    if (!lastGitID) {
-      lastGitID = issues[0].number;
-    }
-    for (const i of issues) {
-      if (i.number !== gitID) {
-        issueList.push(i);
-      } else {
-        isIEF = false;
-        break;
+      for (const permit of permitList) {
+        const { data: userData } = await octokit.request("GET /user/:id", { id: permit.locations.user_id });
```

Comment:
There should be an octokit method for users. Always use the official methods vs making custom requests like this.

Reply:
There is none using the id of the user, only the name. Refer to https://octokit.rest/GET/users/%7Busername%7D